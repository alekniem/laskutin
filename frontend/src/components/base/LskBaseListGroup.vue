<template>
    <v-list-group no-action v-bind="$attrs">
        <template #activator>
            <v-list-item-content
                :id="activatorId"
                class="text-caption blue--text text--darken-2 font-weight-bold"
            >
                {{ title | trans }}
            </v-list-item-content>
        </template>
        <LskBaseListItem
            v-for="(itemData, itemId) in items"
            :itemId="itemId"
            :key="itemId"
            :parentId="parentId"
            :text="itemData.text"
            :value="itemData.value"
        />
    </v-list-group>
</template>

<script>
export default {
    name: "LskBaseListGroup",

    props: {
        activatorId: { type: String, required: true },
        items: { type: Object, required: true },
        parentId: { type: String, required: true },
        title: { type: String, required: true },
    },
};
</script>
