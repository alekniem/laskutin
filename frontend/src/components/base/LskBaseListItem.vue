<template>
    <v-list-item :id="listItemId">
        <v-list-item-content
            class="text-caption blue--text text--darken-2 font-weight-bold"
        >
            {{ text | trans }}
        </v-list-item-content>
        <v-list-item-content>
            {{ value }}
        </v-list-item-content>
    </v-list-item>
</template>

<script>
import camelCase from "lodash/camelCase";
import upperFirst from "lodash/upperFirst";

export default {
    name: "LskBaseListItem",

    props: {
        itemId: { type: String, required: true },
        parentId: { type: String, required: true },
        text: { type: String, required: true },
        value: { type: [String, Number], required: true },
    },

    computed: {
        listItemId() {
            return (
                this.parentId + upperFirst(camelCase(this.itemId)) + "ListItem"
            );
        },
    },
};
</script>
