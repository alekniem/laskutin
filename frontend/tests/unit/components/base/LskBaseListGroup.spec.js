import LskBaseListGroup from "@/components/base/LskBaseListGroup";
import LskBaseListItem from "@/components/base/LskBaseListItem";
import { createWrapper } from "UT/lsk_test_helper";

const stubs = {
    LskBaseListItem,
};

describe("LskBaseListGroup", () => {
    it("should match snapshot", () => {
        const wrapper = createWrapper(LskBaseListGroup, {
            propsData: {
                activatorId: "TEST_ACTIVATOR_ID",
                items: {},
                parentId: "TEST_PARENT_ID",
                title: "TEST TITLE",
            },
        });
        expect(wrapper.html()).toMatchSnapshot("no items, group collapsed");
    });

    it("should match snapshot", () => {
        const wrapper = createWrapper(LskBaseListGroup, {
            propsData: {
                activatorId: "TEST_ACTIVATOR_ID",
                items: {},
                parentId: "TEST_PARENT_ID",
                title: "TEST TITLE",
                value: true,
            },
            stubs,
        });
        expect(wrapper.html()).toMatchSnapshot("no items, group expanded");
    });

    it("should match snapshot", () => {
        const wrapper = createWrapper(LskBaseListGroup, {
            propsData: {
                activatorId: "TEST_ACTIVATOR_ID",
                items: {
                    test_one: { text: "TEST TEXT ONE", value: "TEST VALUE ONE" },
                },
                parentId: "TEST_PARENT_ID",
                title: "TEST TITLE",
            },
        });
        expect(wrapper.html()).toMatchSnapshot("one item, group collapsed");
    });

    it("should match snapshot", () => {
        const wrapper = createWrapper(LskBaseListGroup, {
            propsData: {
                activatorId: "TEST_ACTIVATOR_ID",
                items: {
                    test_one: { text: "TEST TEXT ONE", value: "TEST VALUE ONE" },
                },
                parentId: "TEST_PARENT_ID",
                title: "TEST TITLE",
                value: true,
            },
            stubs,
        });
        expect(wrapper.html()).toMatchSnapshot("one item, group expanded");
    });

    it("should match snapshot", () => {
        const wrapper = createWrapper(LskBaseListGroup, {
            propsData: {
                activatorId: "TEST_ACTIVATOR_ID",
                items: {
                    test_one: { text: "TEST TEXT ONE", value: "TEST VALUE ONE" },
                    test_two: { text: "TEST TEXT TWO", value: "TEST VALUE TWO" },
                },
                parentId: "TEST_PARENT_ID",
                title: "TEST TITLE",
            },
        });
        expect(wrapper.html()).toMatchSnapshot("two items, group collapsed");
    });

    it("should match snapshot", () => {
        const wrapper = createWrapper(LskBaseListGroup, {
            propsData: {
                activatorId: "TEST_ACTIVATOR_ID",
                items: {
                    test_one: { text: "TEST TEXT ONE", value: "TEST VALUE ONE" },
                    test_two: { text: "TEST TEXT TWO", value: "TEST VALUE TWO" },
                },
                parentId: "TEST_PARENT_ID",
                title: "TEST TITLE",
                value: true,
            },
            stubs,
        });
        expect(wrapper.html()).toMatchSnapshot("two items, group expanded");
    });
});
