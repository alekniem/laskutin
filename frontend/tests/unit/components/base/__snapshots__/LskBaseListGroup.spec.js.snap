// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LskBaseListGroup should match snapshot: no items, group collapsed 1`] = `
<div class="v-list-group v-list-group--no-action">
  <div tabindex="0"
       aria-expanded="false"
       role="button"
       class="v-list-group__header v-list-item v-list-item--link theme--light"
  >
    <div id="TEST_ACTIVATOR_ID"
         class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold"
    >
      TEST TITLE
    </div>
    <div class="v-list-item__icon v-list-group__header__append-icon">
      <i aria-hidden="true"
         class="v-icon notranslate mdi mdi-chevron-down theme--light"
      >
      </i>
    </div>
  </div>
  <transition-stub name="expand-transition">
  </transition-stub>
</div>
`;

exports[`LskBaseListGroup should match snapshot: no items, group expanded 1`] = `
<div class="v-list-group v-list-group--active v-list-group--no-action primary--text"
     value="true"
>
  <div tabindex="0"
       aria-expanded="true"
       role="button"
       class="v-list-group__header v-list-item v-list-item--active v-list-item--link theme--light"
  >
    <div id="TEST_ACTIVATOR_ID"
         class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold"
    >
      TEST TITLE
    </div>
    <div class="v-list-item__icon v-list-group__header__append-icon">
      <i aria-hidden="true"
         class="v-icon notranslate mdi mdi-chevron-down theme--light"
      >
      </i>
    </div>
  </div>
  <transition-stub name="expand-transition">
    <div class="v-list-group__items">
    </div>
  </transition-stub>
</div>
`;

exports[`LskBaseListGroup should match snapshot: one item, group collapsed 1`] = `
<div class="v-list-group v-list-group--no-action">
  <div tabindex="0"
       aria-expanded="false"
       role="button"
       class="v-list-group__header v-list-item v-list-item--link theme--light"
  >
    <div id="TEST_ACTIVATOR_ID"
         class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold"
    >
      TEST TITLE
    </div>
    <div class="v-list-item__icon v-list-group__header__append-icon">
      <i aria-hidden="true"
         class="v-icon notranslate mdi mdi-chevron-down theme--light"
      >
      </i>
    </div>
  </div>
  <transition-stub name="expand-transition">
  </transition-stub>
</div>
`;

exports[`LskBaseListGroup should match snapshot: one item, group expanded 1`] = `
<div class="v-list-group v-list-group--active v-list-group--no-action primary--text"
     value="true"
>
  <div tabindex="0"
       aria-expanded="true"
       role="button"
       class="v-list-group__header v-list-item v-list-item--active v-list-item--link theme--light"
  >
    <div id="TEST_ACTIVATOR_ID"
         class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold"
    >
      TEST TITLE
    </div>
    <div class="v-list-item__icon v-list-group__header__append-icon">
      <i aria-hidden="true"
         class="v-icon notranslate mdi mdi-chevron-down theme--light"
      >
      </i>
    </div>
  </div>
  <transition-stub name="expand-transition">
    <div class="v-list-group__items">
      <div tabindex="-1"
           id="TEST_PARENT_IDTestOneListItem"
           class="v-list-item theme--light"
      >
        <div class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold">
          TEST TEXT ONE
        </div>
        <div class="v-list-item__content">
          TEST VALUE ONE
        </div>
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`LskBaseListGroup should match snapshot: two items, group collapsed 1`] = `
<div class="v-list-group v-list-group--no-action">
  <div tabindex="0"
       aria-expanded="false"
       role="button"
       class="v-list-group__header v-list-item v-list-item--link theme--light"
  >
    <div id="TEST_ACTIVATOR_ID"
         class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold"
    >
      TEST TITLE
    </div>
    <div class="v-list-item__icon v-list-group__header__append-icon">
      <i aria-hidden="true"
         class="v-icon notranslate mdi mdi-chevron-down theme--light"
      >
      </i>
    </div>
  </div>
  <transition-stub name="expand-transition">
  </transition-stub>
</div>
`;

exports[`LskBaseListGroup should match snapshot: two items, group expanded 1`] = `
<div class="v-list-group v-list-group--active v-list-group--no-action primary--text"
     value="true"
>
  <div tabindex="0"
       aria-expanded="true"
       role="button"
       class="v-list-group__header v-list-item v-list-item--active v-list-item--link theme--light"
  >
    <div id="TEST_ACTIVATOR_ID"
         class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold"
    >
      TEST TITLE
    </div>
    <div class="v-list-item__icon v-list-group__header__append-icon">
      <i aria-hidden="true"
         class="v-icon notranslate mdi mdi-chevron-down theme--light"
      >
      </i>
    </div>
  </div>
  <transition-stub name="expand-transition">
    <div class="v-list-group__items">
      <div tabindex="-1"
           id="TEST_PARENT_IDTestOneListItem"
           class="v-list-item theme--light"
      >
        <div class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold">
          TEST TEXT ONE
        </div>
        <div class="v-list-item__content">
          TEST VALUE ONE
        </div>
      </div>
      <div tabindex="-1"
           id="TEST_PARENT_IDTestTwoListItem"
           class="v-list-item theme--light"
      >
        <div class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold">
          TEST TEXT TWO
        </div>
        <div class="v-list-item__content">
          TEST VALUE TWO
        </div>
      </div>
    </div>
  </transition-stub>
</div>
`;
